File: src/main/java/com/matag/game/player/Library.java
package com.matag.game.player;

import java.util.Collections;
import java.util.List;

import org.springframework.context.annotation.Scope;
import org.springframework.stereotype.Component;

import com.matag.game.cardinstance.CardInstance;
import com.matag.game.cardinstance.CardListComponent;

@Component
@Scope("prototype")
public class Library extends CardListComponent {
  public List<CardInstance> peek(int n) {
    return this.cards.subList(0, n);
  }

  public CardInstance draw() {
    return this.cards.remove(0);
  }

  public Library shuffle() {
    Collections.shuffle(cards);
    return this;
  }
}

File: src/main/kotlin/com/matag/game/player/Library.kt
package com.matag.game.player

import com.matag.game.cardinstance.CardInstance
import com.matag.game.cardinstance.CardListComponent

class Library : CardListComponent() {
    fun peek(n: Int): List<CardInstance> =
        cards.subList(0, n)

    fun draw(): CardInstance =
        cards.first().also {
            cards.drop(1)
        }

    fun shuffle() {
        cards = cards.shuffled()
    }
}
